src/py/reactpy/reactpy/core/layout.py

```diff
@@ -416,10 +455,25 @@ def _unmount_model_states(self, old_states: list[_ModelState]) -> None:
             if model_state.is_component_state:
                 life_cycle_state = model_state.life_cycle_state
                 del self._model_states_by_life_cycle_state_id[life_cycle_state.id]
-                life_cycle_state.hook.affect_component_will_unmount()
+                await life_cycle_state.hook.affect_component_will_unmount()
```

Comment:
`affect_component_will_unmount` has always been a bit unclear naming to me. I think it's the word `affect` that has always thrown me off.

Reply:
It's definitely an odd naming convention. I think I was trying to indicate what the life cycle methods were when I originally did it. Definitely something that could be changed in a re-write of `Layout`'s internals.